datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/knowledge/UpdateDocumentSettingsResolver.java

```diff
@@ -0,0 +1,66 @@
+package com.linkedin.datahub.graphql.resolvers.knowledge;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.bindArgument;
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.authorization.AuthorizationUtils;
+import com.linkedin.datahub.graphql.concurrency.GraphQLConcurrencyUtils;
+import com.linkedin.datahub.graphql.exception.AuthorizationException;
+import com.linkedin.datahub.graphql.generated.UpdateDocumentSettingsInput;
+import com.linkedin.metadata.service.DocumentService;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Resolver used for updating the settings of a Document on DataHub. Requires the EDIT_ENTITY_DOCS
+ * or EDIT_ENTITY privilege for the document or MANAGE_DOCUMENTS privilege.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class UpdateDocumentSettingsResolver implements DataFetcher<CompletableFuture<Boolean>> {
+
+  private final DocumentService _documentService;
+
+  @Override
+  public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throws Exception {
+
+    final QueryContext context = environment.getContext();
+    final UpdateDocumentSettingsInput input =
+        bindArgument(environment.getArgument("input"), UpdateDocumentSettingsInput.class);
+
+    final Urn documentUrn = UrnUtils.getUrn(input.getUrn());
+
+    return GraphQLConcurrencyUtils.supplyAsync(
+        () -> {
+          // Use the same authorization check as update operations - need to edit the document
+          if (!AuthorizationUtils.canEditDocument(documentUrn, context)) {
+            throw new AuthorizationException(
+                "Unauthorized to perform this action. Please contact your DataHub administrator.");
+          }
+
+          try {
+            _documentService.updateDocumentSettings(
```

Comment:
New resolver to enable you to add the document to the global context sidebar after creation (if you'd like)